query-service-impl/build.gradle.kts

```diff
@@ -22,11 +22,23 @@ dependencies {
     implementation("io.netty:netty-handler:4.1.71.Final") {
       because("Multiple vulnerabilities")
     }
+    implementation("org.jetbrains.kotlin:kotlin-stdlib:1.6.0") {
+      because(
+        "Improper Locking [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGJETBRAINSKOTLIN-2628385] " +
+          "in org.jetbrains.kotlin:kotlin-stdlib@1.4.10"
+      )
+    }
+    testImplementation("junit:junit:4.13.1") {
```

Comment:
this doesn't seem right, we don't use junit 4 do we? Is something bringing it in transitively?

Reply:
was not required when I pulled this down and upgraded locally:


```diff

diff --git a/query-service-api/build.gradle.kts b/query-service-api/build.gradle.kts
index f4e0248..75bacc2 100644
--- a/query-service-api/build.gradle.kts
+++ b/query-service-api/build.gradle.kts
@@ -23,7 +23,7 @@ protobuf {
     // the identifier, which can be referred to in the "plugins"
     // container of the "generateProtoTasks" closure.
     id("grpc_java") {
-      artifact = "io.grpc:protoc-gen-grpc-java:1.43.1"
+      artifact = "io.grpc:protoc-gen-grpc-java:1.45.1"
     }
 
     if (generateLocalGoGrpcFiles) {
@@ -66,17 +66,11 @@ tasks.test {
 }
 
 dependencies {
-  api(platform("io.grpc:grpc-bom:1.43.1"))
+  api(platform("io.grpc:grpc-bom:1.45.1"))
   api("io.grpc:grpc-protobuf")
   api("io.grpc:grpc-stub")
   api("javax.annotation:javax.annotation-api:1.3.2")
 
-  constraints {
-    implementation("com.google.protobuf:protobuf-java:3.19.2") {
-      because("https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-2331703")
-    }
-  }
-
   testImplementation("org.junit.jupiter:junit-jupiter:5.7.1")
   testImplementation("com.google.protobuf:protobuf-java-util:3.19.2")
 }
diff --git a/query-service-client/build.gradle.kts b/query-service-client/build.gradle.kts
index b7529cd..84f6c0c 100644
--- a/query-service-client/build.gradle.kts
+++ b/query-service-client/build.gradle.kts
@@ -7,7 +7,7 @@ plugins {
 
 dependencies {
   api(project(":query-service-api"))
-  implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.7.0")
+  implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.7.2")
 
   // Logging
   implementation("org.slf4j:slf4j-api:1.7.32")
diff --git a/query-service-impl/build.gradle.kts b/query-service-impl/build.gradle.kts
index 05c13c3..e727648 100644
--- a/query-service-impl/build.gradle.kts
+++ b/query-service-impl/build.gradle.kts
@@ -22,12 +22,15 @@ dependencies {
     implementation("io.netty:netty-handler:4.1.71.Final") {
       because("Multiple vulnerabilities")
     }
+    implementation("org.jetbrains.kotlin:kotlin-stdlib:1.6.21") {
+      because("https://snyk.io/vuln/SNYK-JAVA-ORGJETBRAINSKOTLIN-2628385")
+    }
   }
   api(project(":query-service-api"))
   api("com.typesafe:config:1.4.1")
-  implementation("org.hypertrace.core.grpcutils:grpc-context-utils:0.7.0")
-  implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.7.0")
-  implementation("org.hypertrace.core.grpcutils:grpc-server-rx-utils:0.7.0")
+  implementation("org.hypertrace.core.grpcutils:grpc-context-utils:0.7.2")
+  implementation("org.hypertrace.core.grpcutils:grpc-client-utils:0.7.2")
+  implementation("org.hypertrace.core.grpcutils:grpc-server-rx-utils:0.7.2")
   implementation("org.hypertrace.core.attribute.service:attribute-service-api:0.12.3")
   implementation("org.hypertrace.core.attribute.service:attribute-projection-registry:0.12.3")
   implementation("org.hypertrace.core.attribute.service:caching-attribute-service-client:0.12.3")
@@ -41,9 +44,9 @@ dependencies {
   implementation("commons-codec:commons-codec:1.15")
   implementation("org.hypertrace.core.serviceframework:platform-metrics:0.1.33")
   implementation("com.google.protobuf:protobuf-java-util:3.19.2")
-  implementation("com.google.guava:guava:30.1.1-jre")
+  implementation("com.google.guava:guava:31.1-jre")
   implementation("io.reactivex.rxjava3:rxjava:3.0.11")
-  implementation("com.squareup.okhttp3:okhttp:4.9.1")
+  implementation("com.squareup.okhttp3:okhttp:4.9.3")
   annotationProcessor("org.projectlombok:lombok:1.18.20")
   compileOnly("org.projectlombok:lombok:1.18.20")
 
@@ -52,5 +55,5 @@ dependencies {
   testImplementation("org.mockito:mockito-core:3.8.0")
   testImplementation("org.mockito:mockito-junit-jupiter:3.8.0")
   testImplementation("org.apache.logging.log4j:log4j-slf4j-impl:2.17.1")
-  testImplementation("com.squareup.okhttp3:mockwebserver:4.9.1")
+  testImplementation("com.squareup.okhttp3:mockwebserver:4.9.3")
 }
diff --git a/query-service/build.gradle.kts b/query-service/build.gradle.kts
index c2f31fd..7a1c9ab 100644
--- a/query-service/build.gradle.kts
+++ b/query-service/build.gradle.kts
@@ -10,7 +10,7 @@ plugins {
 
 dependencies {
   implementation(project(":query-service-impl"))
-  implementation("org.hypertrace.core.grpcutils:grpc-server-utils:0.7.0")
+  implementation("org.hypertrace.core.grpcutils:grpc-server-utils:0.7.2")
   implementation("org.hypertrace.core.serviceframework:platform-service-framework:0.1.33")
   implementation("org.slf4j:slf4j-api:1.7.32")
   implementation("com.typesafe:config:1.4.1")
@@ -28,7 +28,7 @@ dependencies {
   integrationTestImplementation("org.apache.kafka:kafka-clients:5.5.1-ccs")
   integrationTestImplementation("org.apache.kafka:kafka-streams:5.5.1-ccs")
   integrationTestImplementation("org.apache.avro:avro:1.11.0")
-  integrationTestImplementation("com.google.guava:guava:30.1.1-jre")
+  integrationTestImplementation("com.google.guava:guava:31.1-jre")
   integrationTestImplementation("org.hypertrace.core.datamodel:data-model:0.1.12")
   integrationTestImplementation("org.hypertrace.core.kafkastreams.framework:kafka-streams-serdes:0.1.13")
 
```

Reply:
> ```diff
> com.squareup.okhttp3:mockwebserver:4.9.3
> ```

It was transitive from `com.squareup.okhttp3:mockwebserver:4.9.3`. Let me try other grpc ones that you suggested.

Reply:
Done